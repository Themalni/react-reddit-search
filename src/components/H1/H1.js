import styled from 'styled-components'

const H1 = styled.h1`
  margin-top: 3em;
  font-size:  ${({ isLarge }) => isLarge ? '2.5em' : '2em'};
  color:  ${(props) => props.theme.colors.white}
  font-weight: ${(props) => props.theme.font.bold}
`

export default H1