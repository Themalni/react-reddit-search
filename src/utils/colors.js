export const colors = {
  white: '#fff',
  graphite: '#666',
  grey: '#8a96a3',
  yellowOrange: '#ff9933',
  orange: '#ed5f55',
  orangeDark: '#d8574e',
  blue: '#3a81b7',
  blueLight: '#64a0ce'
};

// d02216