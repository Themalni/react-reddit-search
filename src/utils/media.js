export const media = {
  phone: '@media (min-width: 320px)',
  tablet: '@media (min-width: 728px)',
  laptop: '@media (min-width: 1024px)',
  desktop: '@media (min-width: 1280px)'
}