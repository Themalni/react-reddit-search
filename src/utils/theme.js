import { colors } from './colors';

export const theme = {
  colors,
  font: {
    thin: 300,
    regular: 400,
    bold: 700,
  },
  media: {
    phone: '@media (min-width: 320px)',
    tablet: '@media (min-width: 728px)',
    laptop: '@media (min-width: 1024px)',
    desktop: '@media (min-width: 1280px)'
  }
}