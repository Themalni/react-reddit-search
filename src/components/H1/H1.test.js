import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { theme } from '../../utils/theme';
import 'jest-styled-components'
import H1 from './H1';

describe('H1', () => {

  it("renders correctly", () => {
    const { container } = render(<H1 theme={theme}></H1>)
    expect(container.firstChild).toMatchSnapshot();
  })

})