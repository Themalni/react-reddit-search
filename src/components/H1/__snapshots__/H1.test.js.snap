// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 renders correctly 1`] = `
.c0 {
  margin-top: 3em;
  font-size: 2em;
  color: #fff;
  font-weight: 700;
}

@media (min-width:728px) {
  .c0 {
    font-size: 2.5em;
  }
}

<h1
  class="c0"
/>
`;
